File: Python/Stacks & Queue/L_maxSlidingWindow.py
from collections import deque


def maxSlidingWindow( nums, k):
        op = []
        q = deque()
        l=r=0
        while r < len(nums):
            while q and nums[q[-1]] < nums[r]:
                q.pop()
            q.append(r)
            if l > q[0]:
                q.popleft()
            if (r+1) >= k:
                op.append(nums[q[0]])
                l+=1
            r+=1
        return op

File: cpp/Array/L_MaxSlidingWIndow.cpp
#include<bits/stdc++.h>
using namespace std;

vector<int> maxSlidingWindow(vector<int>& nums, int k) {
    deque<int> queue;
    vector<int> result;
    int left = 0, right = 0;
    while ( right < nums.size()) {
        while (!queue.empty() && nums[queue.back()] < nums[right]) queue.pop_back();
        queue.push_back(right);
        if (left > queue.front()) queue.pop_front();
        if (right + 1 >= k) {
            result.push_back(nums[queue.front()]);
            left++;
        }
        right++;
    }
    return result;
}

int main() {

}